controllers/machinedeployment_sync.go

```diff
@@ -406,55 +403,41 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 
 func (r *MachineDeploymentReconciler) scaleMachineSet(ctx context.Context, ms *clusterv1.MachineSet, newScale int32, deployment *clusterv1.MachineDeployment) error {
 	if ms.Spec.Replicas == nil {
-		return errors.Errorf("spec replicas for machine set %v is nil, this is unexpected", ms.Name)
+		return errors.Errorf("spec replicas for MachineSet %s is nil, this is unexpected", client.ObjectKeyFromObject(ms).String())
 	}
 
-	// No need to scale
-	if *(ms.Spec.Replicas) == newScale {
-		return nil
-	}
-
-	var scalingOperation string
-	if *(ms.Spec.Replicas) < newScale {
-		scalingOperation = "up"
-	} else {
-		scalingOperation = "down"
-	}
-
-	return r.scaleMachineSetOperation(ctx, ms, newScale, deployment, scalingOperation)
-}
-
-func (r *MachineDeploymentReconciler) scaleMachineSetOperation(ctx context.Context, ms *clusterv1.MachineSet, newScale int32, deployment *clusterv1.MachineDeployment, scaleOperation string) error {
-	if ms.Spec.Replicas == nil {
-		return errors.Errorf("spec replicas for machine set %v is nil, this is unexpected", ms.Name)
-	}
-
-	sizeNeedsUpdate := *(ms.Spec.Replicas) != newScale
-
 	annotationsNeedUpdate := mdutil.ReplicasAnnotationsNeedUpdate(
 		ms,
 		*(deployment.Spec.Replicas),
 		*(deployment.Spec.Replicas)+mdutil.MaxSurge(*deployment),
 	)
 
-	if sizeNeedsUpdate || annotationsNeedUpdate {
-		patchHelper, err := patch.NewHelper(ms, r.Client)
-		if err != nil {
-			return err
-		}
+	// No need to scale nor setting annotations.
+	if *(ms.Spec.Replicas) == newScale && !annotationsNeedUpdate {
+		return nil
+	}
+
+	patchHelper, err := patch.NewHelper(ms, r.Client)
+	if err != nil {
+		return err
+	}
 
-		*(ms.Spec.Replicas) = newScale
-		mdutil.SetReplicasAnnotations(ms, *(deployment.Spec.Replicas), *(deployment.Spec.Replicas)+mdutil.MaxSurge(*deployment))
+	// Save original replicas to log in event.
+	originalReplicas := *(ms.Spec.Replicas)
 
-		err = patchHelper.Patch(ctx, ms)
-		if err != nil {
-			r.recorder.Eventf(deployment, corev1.EventTypeWarning, "FailedScale", "Failed to scale MachineSet %q: %v", ms.Name, err)
-		} else if sizeNeedsUpdate {
-			r.recorder.Eventf(deployment, corev1.EventTypeNormal, "SuccessfulScale", "Scaled %s MachineSet %q to %d", scaleOperation, ms.Name, newScale)
-		}
+	// Mutate.
+	ms.Spec.Replicas = &newScale
+	mdutil.SetReplicasAnnotations(ms, *(deployment.Spec.Replicas), *(deployment.Spec.Replicas)+mdutil.MaxSurge(*deployment))
+
+	if err = patchHelper.Patch(ctx, ms); err != nil {
+		r.recorder.Eventf(deployment, corev1.EventTypeWarning, "FailedScale", "Failed to scale MachineSet %s: %v",
+			client.ObjectKeyFromObject(ms).String(), err)
```

Comment:
```suggestion
		r.recorder.Eventf(deployment, corev1.EventTypeWarning, "FailedScale", "Failed to scale MachineSet %v: %v",
			client.ObjectKeyFromObject(ms), err)
```

Reply:
Are you suggesting to drop `.String()` and record the objectKey struct here?